### PR TITLE
fix: handle default value in Multiselect and select widget

### LINKED ISSUE #12225: [Bug]-[300]:Default value is displayed on widget even when it is not available in drop-down options
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://watch.screencastify.com/v/rTmvAKEDYT6lAqffXIXq

### Steps To Reproduce

1. Drag and drop a Select/multi-select/TreeSelect/Multi tree select
2. Add any random value in Default Value
3. Observe that it is shown in widget even when the default value is not present in drop-down options

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> fix issue of default value is displayed on  Select/multi-select/TreeSelect/Multi tree select widget even it is not available in widget options list

Fixes #12225

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/issue-12225 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :red_circle: | app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx | 73 **(-0.74)** | 48.57 **(0)** | 57.14 **(0)** | 72.63 **(-0.77)**
 :red_circle: | app/client/src/widgets/SelectWidget/widget/index.tsx | 51.81 **(-0.63)** | 19.12 **(-1.85)** | 42.11 **(0)** | 52.5 **(-0.66)**</details>

### CODE REVIEW

File: app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -539,11 +540,13 @@ class MultiSelectWidget extends BaseWidget<
...
     const minDropDownWidth = MinimumPopupRows * this.props.parentColumnSpace;
     const { componentWidth } = this.getComponentDimensions();
-    const values: LabelValueType[] = this.props.selectedOptions
+    let values: LabelValueType[] = this.props.selectedOptions
       ? this.props.selectedOptions.map((o) =>
           isString(o) || isNumber(o) ? { value: o } : { value: o.value },
         )
       : [];
+    values = intersectionBy(options, values, "value");
+
```
Comment (ashit-rath) [inline app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx:549]: This solution wouldn't resolve the issue completely.
The `selectedOptions` should always have the correct options from the start and don't need to be filtered explicitly by using `intersectionBy` .
The reason for this is, as the `selectedOptionValue` is derived from `selectedOptionValueArr` and these two are used to compute other values (`selectedOptionLabels`, `selectedOptionValues`). If we have to explicitly filter before using then we would have to repeat this logic everywhere.
So my recommendation is to fix it at the source i.e `selectedOptionValueArr` and everything would fall in order.

File: app/client/src/widgets/SelectWidget/widget/index.tsx [hunk 2]
```diff
@@ -444,6 +444,8 @@ class SelectWidget extends BaseWidget<SelectWidgetProps, WidgetState> {
     const selectedIndex = findIndex(this.props.options, {
       value: this.props.selectedOptionValue,
     });
+    // show value in select input if serverSideFiltering enable or value exist in options
+    const showValue = this.props.serverSideFiltering || selectedIndex !== -1;
```
Comment (ashit-rath) [inline app/client/src/widgets/SelectWidget/widget/index.tsx:448]: You would have to alter the logic of your solution here as well
`selectedOptionLabel` and `selectedOptionValue` would still show that value

**Review by ashit-rath — CHANGES_REQUESTED** [review]
Please update the title of the PR to give more context and add cypress tests.

**Comment by techbhavin** [comment]
@somangshu  @dilippitchika  @ashit-rath 

closing this PR. a related issue is covered is PR[#13079](https://github.com/appsmithorg/appsmith/pull/13079)

**Pull request closed without merging**